base/testfiles-lthooks2/ltpara-008.tlg

```diff
@@ -25,14 +25,20 @@ PARA: 1 end   (document level on input line ...)
 Completed box being shipped out [1]
 \vbox(633.0+0.0)x407.0
 .\hbox(0.0+0.0)x0.0
+..\kern -72.26999
+..\vbox(0.0+0.0)x0.0, glue set 72.26999fil
+...\kern -72.26999
+...\hbox(0.0+0.0)x0.0
+....\glue 0.0 plus 1.0fil minus 1.0fil
+...\glue 0.0 plus 1.0fil minus 1.0fil
 .\glue 16.0
```

Comment:
does anybody understands that difference? It doesn't hurt but it is a construction that looks very unnecessary, ie no content but boxes and glue put together. Happens in a couple of places.

Reply:
It's the page colour code in `pdftex.def`, but I'm not sure why it shows as `\ifGPT@pagecolor` should be false.

Reply:
then perhaps that is worth investigating rather than accepting just because it doesn't hurt (in this case), no?

Reply:
You get that as soon something is in the shipout hooks. E.g. 
~~~~
\documentclass{article}
\showoutput
\AddToHook{shipout/background}{\relax}
\begin{document}
aaa
\end{document}
~~~~

Reply:
Yes, it looks about right for the "background box".

Reply:
so it does, but why is there an (empty) background box if pdftex is used but not if etex was used?

Reply:
because pdftex.def puts code into `\@kernel@before@shipout@background` (for the pagecolor), and any non empty content triggers the box too: 

~~~~
\documentclass{article}

\showoutput 
\makeatletter 
\def\@kernel@before@shipout@background{\iffalse\fi}
\makeatother
\begin{document}
blub
\end{document}
~~~~

Reply:
well the code in pdftex.def is a bit more complicated, but it is the same principle: as soon as you add some code to the hook the box for the shipout/background is created, even if the code at the end expands to nothingness.